### PR TITLE
[Docs] Add table of contents and introduction section

### PR DESCRIPTION
**Pull request by fzdarsky**

### CODE REVIEW

File: docs/user/getting-started.md [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+    flightctl oci://quay.io/flightctl/flightctl-helm
+```
+
+## Installing the Flight Control CLI
+
+On RPM-based systems, install the `flightctl` CLI using
+
+```console
+sudo dnf copr enable -y @redhat-et/flightctl-dev
+sudo dnf install -y flightctl
```
Comment (oglok) [inline docs/user/getting-started.md:34]: we don't build RPMs for the CLI (yet)

**Comment by fzdarsky** [comment]
I'm factoring out the "Getting Started" into a separate PR to streamline the review process.

File: docs/user/introduction.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+## Project Vision
+
+Flight Control's goal is to provide simple, scalable, and secure management of edge devices and applications. Users declare the operating system version, host configuration, and set of applications they want to run on an individual device or a whole fleets of devices and Flight Control will roll out the target configuration to devices where a device agent will automatically apply them and report progress and health status back up.
+
+Flight Control is designed for modern, container-centric toolchains and operational best-practices and works best on image-based Linux operating systems running bootc or ostree and with container workloads running on Podman/Docker or Kubernetes.
+
+Features and use cases Flight Control aims to support:
+
+* Declarative APIs well-suited for a GitOps lifestyle.
```
Comment (mangelajo) [inline docs/user/introduction.md:11]:
```suggestion
* Declarative APIs well-suited for a GitOps management.
```

File: docs/user/introduction.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+  * APIs are Kubernetes-like, so should instantly feel familiar to Kubernetes users and allow them to reuse tools and toolchains.
+  * APIs are not depending on Kubernetes, though, and target a shallow learning curve for non-Kubernetes users.
+* Web UI to manage and monitor devices and applications for a ClickOps lifestyle.
```
Comment (mangelajo) [inline docs/user/introduction.md:14]:
```suggestion
* Web UI to manage and monitor devices and applications for ClickOps management.
```

File: docs/user/introduction.md [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+**Agent** - The Flight Control Agent runs on each device and is responsible for securely enrolling into the Service, querying the Service for updates, autonomously applying these updates, and reporting status on the updates and the health of devices back to the Service.
+
+## High-Level Architecture
+
+Flight Control consists of the highlighted components show in the following high-level architecture diagram:
+
+![no-alt-text](../images/flightctl-highlevel-architecture.svg#only-light)
+![no-alt-text](../images/flightctl-highlevel-architecture-dark.svg#only-dark)
+
+The Flight Control Service consists of an API server, worker processes (not shown), and a PostgreSQL database for storing inventory and runtime target and reported states. The API server exposes two endpoints:
```
Comment (mangelajo) [inline docs/user/introduction.md:53]: It's actually three with gRPC now, although probably we want the end state to just be two.

File: docs/user/introduction.md [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+
+The Flight Control Service talks to various external systems to authenticate and authorize users, get mTLS certificates signed, or query configuration for managed devices.
+
+The Flight Control Agent is a process running on each managed device. It always "calls home" to the Service, so the device can be on a private network or have a dynamic IP address. The agent handles the enrollment process with the service, then periodically polls the Service for a new target configuration. It also periodically sends a heartbeat to the Service and notifies the Service when the device or application status change relative to the desired target configuration.
+
+When the Agent receives a new target configuration from the Service, it
+
+1. downloads all required assets (OS image, application container images, etc.) over the network to disk, so it doesn't depend on network connectivity during the update,
```
Comment (mangelajo) [inline docs/user/introduction.md:64]: looks reasonable for a final goal :)

**Review by mangelajo — COMMENTED** [review]
just some nits

File: docs/user/introduction.md (hunk 2 above)
Comment (avishayt) [reply docs/user/introduction.md:11]:
```suggestion
* Declarative APIs well-suited for GitOps management.
```

File: docs/user/introduction.md [hunk 6]
```diff
@@ -0,0 +1,72 @@
+# Introduction
+
+## Project Vision
+
+Flight Control's goal is to provide simple, scalable, and secure management of edge devices and applications. Users declare the operating system version, host configuration, and set of applications they want to run on an individual device or a whole fleets of devices and Flight Control will roll out the target configuration to devices where a device agent will automatically apply them and report progress and health status back up.
+
+Flight Control is designed for modern, container-centric toolchains and operational best-practices and works best on image-based Linux operating systems running bootc or ostree and with container workloads running on Podman/Docker or Kubernetes.
```
Comment (avishayt) [inline docs/user/introduction.md:7]:
```suggestion
Flight Control is designed for modern, container-centric toolchains and operational best practices. It works best on image-based Linux operating systems running bootc or ostree and with container workloads running on Podman/Docker or Kubernetes.
```

File: docs/user/introduction.md [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+
+Features and use cases Flight Control aims to support:
+
+* Declarative APIs well-suited for a GitOps lifestyle.
+  * APIs are Kubernetes-like, so should instantly feel familiar to Kubernetes users and allow them to reuse tools and toolchains.
+  * APIs are not depending on Kubernetes, though, and target a shallow learning curve for non-Kubernetes users.
```
Comment (avishayt) [inline docs/user/introduction.md:13]:
```suggestion
  * APIs do not depend on Kubernetes, though, and target a shallow learning curve for non-Kubernetes users.
```

File: docs/user/introduction.md [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+**Service** -  The Flight Control Service handles user and agent authentication and authorization, device enrollment and inventory, rolling out updates to devices, and rolling up status from devices.
+
+**Agent** - The Flight Control Agent runs on each device and is responsible for securely enrolling into the Service, querying the Service for updates, autonomously applying these updates, and reporting status on the updates and the health of devices back to the Service.
+
+## High-Level Architecture
+
+Flight Control consists of the highlighted components show in the following high-level architecture diagram:
+
+![no-alt-text](../images/flightctl-highlevel-architecture.svg#only-light)
+![no-alt-text](../images/flightctl-highlevel-architecture-dark.svg#only-dark)
```
Comment (avishayt) [inline docs/user/introduction.md:51]: When viewing this file on github, it shows up as two separate images, one after the other.

File: docs/user/introduction.md (hunk 4 above)
Comment (avishayt) [inline docs/user/introduction.md:53]:
```suggestion
The Flight Control Service consists of an API server, worker processes (not shown), and a PostgreSQL database for storing inventory, runtime targets, and reported states. The API server exposes two endpoints:
```

File: docs/user/introduction.md [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+
+The Flight Control Service consists of an API server, worker processes (not shown), and a PostgreSQL database for storing inventory and runtime target and reported states. The API server exposes two endpoints:
+
+* The *user-facing API endpoint* is for user to connect to, typically from the CLI or web UI. Users on this endpoint need to authenticate with the configured external authentication service to obtain a JWT token, then can use this token when making requests via HTTPS.
```
Comment (avishayt) [inline docs/user/introduction.md:55]:
```suggestion
* The *user-facing API endpoint* is for users to connect to, typically from the CLI or web UI. Users on this endpoint must authenticate with the configured external authentication service to obtain a JWT token. They can then use this token when making requests via HTTPS.
```

File: docs/user/introduction.md [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+* The *agent-facing API endpoint* is for agents to connect to and is mTLS-protected. That is, the service authenticates the device based on its X.509 client certificates. The device's unique certificate is bootstrapped during enrollment based on hardware root-of-trust, meaning the private key is protected by the TPM and so the client certificate cannot be used by another entity. Certificates are automatically rotated before they expire.
+
+The Flight Control Service talks to various external systems to authenticate and authorize users, get mTLS certificates signed, or query configuration for managed devices.
+
+The Flight Control Agent is a process running on each managed device. It always "calls home" to the Service, so the device can be on a private network or have a dynamic IP address. The agent handles the enrollment process with the service, then periodically polls the Service for a new target configuration. It also periodically sends a heartbeat to the Service and notifies the Service when the device or application status change relative to the desired target configuration.
```
Comment (avishayt) [inline docs/user/introduction.md:60]:
```suggestion
The Flight Control Agent is a process running on each managed device. It always "calls home" to the Service, so the device can be on a private network or have a dynamic IP address. The agent handles the enrollment process with the service and periodically polls the Service for a new target configuration. It also periodically sends a heartbeat to the Service and notifies the Service when the device or application status changes relative to the desired target configuration.
```

File: docs/user/introduction.md [hunk 11]
```diff
@@ -0,0 +1,72 @@
...
+
+When the Agent receives a new target configuration from the Service, it
+
+1. downloads all required assets (OS image, application container images, etc.) over the network to disk, so it doesn't depend on network connectivity during the update,
+2. updates the OS image by delegating to bootc (or rpm-ostree),
+3. updates configuration files on the device's file system by overlaying a set of files sent to it by the Service,
+4. if necessary, reboots into the new OS, otherwise signals services to reload the updated configuration, and
+5. updates applications running on Podman or MicroShift by running the necessary commands.
+
+If applying any of these changes fails or the system does not come back online after reboot (detected greenboot health-checks and optionally user-defined logic), the agent will rollback to the previous OS image and configuration.
```
Comment (avishayt) [inline docs/user/introduction.md:70]:
```suggestion
If applying any of these changes fails or the system does not return online after reboot (detected greenboot health checks and optionally user-defined logic), the agent will rollback to the previous OS image and configuration.
```

File: docs/user/introduction.md (hunk 2 above)
Comment (fzdarsky) [reply docs/user/introduction.md:11]: done

File: docs/user/introduction.md (hunk 4 above)
Comment (fzdarsky) [reply docs/user/introduction.md:53]: Do you feel we should add this or too much detail?

File: docs/user/introduction.md (hunk 3 above)
Comment (fzdarsky) [reply docs/user/introduction.md:14]: done

File: docs/user/introduction.md (hunk 6 above)
Comment (fzdarsky) [reply docs/user/introduction.md:7]: done

File: docs/user/introduction.md [hunk 12]
```diff
@@ -0,0 +1,72 @@
+# Introduction
+
+## Project Vision
+
+Flight Control's goal is to provide simple, scalable, and secure management of edge devices and applications. Users declare the operating system version, host configuration, and set of applications they want to run on an individual device or a whole fleets of devices and Flight Control will roll out the target configuration to devices where a device agent will automatically apply them and report progress and health status back up.
```
[the remark this answers is not in the document]
Comment (fzdarsky) [reply docs/user/introduction.md:5]: done

File: docs/user/introduction.md [hunk 13]
```diff
@@ -0,0 +1,72 @@
...
+
+Flight Control is designed for modern, container-centric toolchains and operational best-practices and works best on image-based Linux operating systems running bootc or ostree and with container workloads running on Podman/Docker or Kubernetes.
+
+Features and use cases Flight Control aims to support:
+
+* Declarative APIs well-suited for a GitOps lifestyle.
+  * APIs are Kubernetes-like, so should instantly feel familiar to Kubernetes users and allow them to reuse tools and toolchains.
```
[the remark this answers is not in the document]
Comment (fzdarsky) [reply docs/user/introduction.md:12]: done

File: docs/user/introduction.md (hunk 10 above)
Comment (fzdarsky) [reply docs/user/introduction.md:60]: done

File: docs/user/introduction.md (hunk 11 above)
Comment (fzdarsky) [reply docs/user/introduction.md:70]: done

File: docs/user/introduction.md (hunk 9 above)
Comment (fzdarsky) [reply docs/user/introduction.md:55]: done

File: docs/user/introduction.md (hunk 7 above)
Comment (fzdarsky) [reply docs/user/introduction.md:13]: done

File: docs/user/introduction.md (hunk 8 above)
Comment (fzdarsky) [reply docs/user/introduction.md]: Thanks for catching this. Apparently, the syntax has been deprecated in favour of inline HTML.

**Review by mangelajo — APPROVED** [review]

**Pull request merged**